tools/dependency_summary.py

```diff
@@ -0,0 +1,815 @@
+    "Apache-2.0",
+    # The MIT license is pretty good, because it's short.
+    "MIT",
+    # Creative Commons Zero is the only Creative Commons license that's MPL-comaptible.
+    "CC0-1.0",
+    # BSD and similar licenses are pretty good; the fewer clauses the better.
+    "ISC",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+]
+
+
+# Packages that get pulled into our dependency tree but we know we definitely don't
+# ever build with in practice, typically because they're platform-specific support
+# for platforms we don't actually support.
+EXCLUDED_PACKAGES = set([
+    "fuchsia-cprng",
+    "fuchsia-zircon",
+    "fuchsia-zircon-sys",
+])
+
+# Known metadata for special extra packages that are not managed by cargo.
+EXTRA_PACKAGE_METADATA = {
+    "ext-jna" : {
+        "name": "jna",
+        "repository": "https://github.com/java-native-access/jna",
+        "license": "Apache-2.0",
+        "license_file": "https://raw.githubusercontent.com/java-native-access/jna/master/AL2.0",
+    },
+    "ext-protobuf": {
+        "name": "protobuf",
+        "repository": "https://github.com/protocolbuffers/protobuf",
+        "license": "BSD-3-Clause",
+        "license_file": "https://raw.githubusercontent.com/protocolbuffers/protobuf/master/LICENSE",
+    },
+    "ext-swift-protobuf": {
+        "name": "swift-protobuf",
+        "repository": "https://github.com/apple/swift-protobuf",
+        "license": "Apache-2.0",
+        "license_file": "https://raw.githubusercontent.com/apple/swift-protobuf/master/LICENSE.txt"
+    },
+    "ext-openssl": {
+        "name": "openssl",
+        "repository": "https://www.openssl.org/source/",
+        "license": "OpenSSL",
+        "license_file": "https://www.openssl.org/source/license-openssl-ssleay.txt",
+    },
+    "ext-sqlcipher": {
+        "name": "sqlcipher",
+        "repository": "https://github.com/sqlcipher/sqlcipher",
+        "license": "BSD-3-Clause",
+        "license_file": "https://raw.githubusercontent.com/sqlcipher/sqlcipher/master/LICENSE",
+    },
+}
+
+# And these are rust packages that pull in the above dependencies.
+# Others are added on a per-target basis during dependency resolution.
+PACKAGES_WITH_EXTRA_DEPENDENCIES = {
+    "openssl-sys": ["ext-openssl"],
+    "ring": ["ext-openssl"],
+    # As a special case, we know that the "logins" crate is the only thing that enables SQLCipher.
+    # In a future iteration we could check the cargo build-plan output to see whether anything is
+    # enabling the sqlcipher feature, but this will do for now.
+    "logins": ["ext-sqlcipher"],
+}
+
+# Hand-audited tweaks to package metadata, for cases where the data given to us by cargo is insufficient.
+# We list both the expected value from `cargo metadata` and the replacement value, to
+# guard against accidentally overwriting future upstream changes in the metadata.
+#
+# Let's try not to add any more dependencies that require us to edit this list!
```

Comment:
I'd feel better about this if it didn't include test/examples/benchmarks/etc.

Reply:
I've succeeded in removing many of the test/example-only dependencies from the list, FWIW.